cc_core.c

```diff
@@ -992,12 +992,93 @@ void general_recursion(FUNCTION f, char* s, char* name, FUNCTION iterate)
 	}
 }
 
+void multiply_by_object_size(void)
+{
+	emit_out("# pointer arithmetic start\n");
+	if((KNIGHT_POSIX == Architecture) || (KNIGHT_NATIVE == Architecture))
+	{
+		emit_out("PUSHR R1 R15\n");
+
+		emit_out("LOADI R1 ");
+		emit_out(int2str(current_target->type->size, 10, FALSE));
+		emit_out("\n");
+
+		emit_out("MULU R0 R1 R0\n");
+		emit_out("POPR R1 R15\n");
+	}
+	else if(X86 == Architecture)
+	{
+		emit_out("push_ebx\n");
+
+		emit_out("mov_ebx, %");
+		emit_out(int2str(current_target->type->size, 10, FALSE));
+		emit_out("\n");
+
+		emit_out("mul_ebx\n");
+		emit_out("pop_ebx\n");
+	}
+	else if(AMD64 == Architecture)
+	{
+		emit_out("push_rbx\n");
+
+		emit_out("mov_rbx, %");
+		emit_out(int2str(current_target->type->size, 10, FALSE));
+		emit_out("\n");
+
+		emit_out("mul_rbx\n");
+		emit_out("pop_rbx\n");
+	}
+	else if(ARMV7L == Architecture)
+	{
+		emit_out("{R1} PUSH_ALWAYS\n");
+
+		emit_out("!0 R1 LOAD32 R15 MEMORY\n~0 JUMP_ALWAYS\n%");
+		emit_out(int2str(current_target->type->size, 10, FALSE));
+		emit_out("\n");
+
+		emit_out("'9' R0 '0' R1 MUL R0 ARITH2_ALWAYS\n");
+		emit_out("{R1} POP_ALWAYS\n");
+	}
+	else if(AARCH64 == Architecture)
+	{
+		emit_out("PUSH_X1\n");
+
+		emit_out("LOAD_W0_AHEAD\nSKIP_32_DATA\n%");
+		emit_out(int2str(current_target->type->size, 10, FALSE));
+		emit_out("\n");
+
+		emit_out("MUL_X0_X1_X0\n");
+		emit_out("POP_X1\n");
+	}
+	else if((RISCV32 == Architecture) || (RISCV64 == Architecture))
+	{
+		if(RISCV32 == Architecture) emit_out("rs1_sp rs2_a1 @-4 sw\n");
+		else if(RISCV64 == Architecture) emit_out("rs1_sp rs2_a1 @-8 sd\n");
+
+		emit_out("rd_a0 !");
+		emit_out(int2str(current_target->type->size, 10, FALSE));
+		emit_out(" addi\n");
+
+		emit_out("rd_a0 rs1_a1 rs2_a0 mul\n");
+
+		if(RISCV32 == Architecture) emit_out("rd_a1 rs1_sp !-4 lw\n");
+		else if(RISCV64 == Architecture) emit_out("rd_a1 rs1_sp !-8 ld\n");
+	}
+
+	emit_out("# pointer arithmetic end\n");
+}
+
 void arithmetic_recursion(FUNCTION f, char* s1, char* s2, char* name, FUNCTION iterate)
 {
 	require(NULL != global_token, "Received EOF in arithmetic_recursion\n");
 	if(match(name, global_token->s))
 	{
 		common_recursion(f);
+		if(name[0] == '+' || name[0] == '-') {
```

Comment:
Can you once again put this on the new line... And lower in this file.

Reply:
Yep, my bad. Should really figure out how to set up my editor to do that. Fixed now.